apps/examples/rdp/rdp_main.c

```diff
@@ -119,16 +125,26 @@ int main(int argc, FAR char *argv[])
 int rdp_main(int argc, char *argv[])
 #endif
 {
-	int pid;
-
 	printf("rdp main!!\n");
 
-	/* Currently TZ support is on flat build only */
-#ifndef CONFIG_BUILD_KERNEL
-	pid = task_create("RDP DEMO", SCHED_PRIORITY_DEFAULT, CONFIG_USERMAIN_STACKSIZE, (main_t)rdp_demo, (FAR char *const *)NULL);
+#ifdef CONFIG_BUILD_FLAT
+	int pid = task_create("RDP DEMO", SCHED_PRIORITY_DEFAULT, CONFIG_USERMAIN_STACKSIZE, (main_t)rdp_demo, (FAR char *const *)NULL);
 	if (pid < 0) {
 		printf("Failed to create rdp demo thread\n");
 	}
+#else
+	int ret = 0;
+	int tc_fd = open(TESTCASE_DRVPATH, O_WRONLY);
+	if (tc_fd < 0) {
+		printf("FAILED to open testcase driver %s\n", TESTCASE_DRVPATH);
+		return;
+	}
+
+	ret = ioctl(tc_fd, TESTIOC_RDP, 0);
+	if (ret	< 0) {
+		printf("ERROR executing RDP test ioctl ret = %d\n", ret);
```

Comment:
ret is always ERROR if ioctl failed. so we should check errno actually.

Reply:
Fixed